rust/src/treefile.rs

```diff
@@ -1905,7 +1905,7 @@ pub(crate) struct DeriveCustom {
 pub(crate) struct DeriveInitramfs {
     pub(crate) regenerate: bool,
     #[serde(skip_serializing_if = "Option::is_none")]
-    pub(crate) etc: Option<Vec<String>>,
+    pub(crate) etc: Option<HashSet<String>>,
```

Comment:
For reproducibility though I think we want `BTreeSet`.

Reply:
Hmm actually though we are using `HashSet` elsewhere.  But particularly with the work on things like the rpmdb normalization for binary reproducbility, because we serialize the treefile to json in `/usr/share/rpm-ostree/treefile.json`, I think we should probably switch to `BTreeSet` right?

Reply:
Hmm actually though we are using `HashSet` elsewhere.  But particularly with the work on things like the rpmdb normalization for binary reproducbility, because we serialize the treefile to json in `/usr/share/rpm-ostree/treefile.json`, I think we should probably switch to `BTreeSet` right?

Reply:
I made it a `HashSet` because I wanted to emphasize that ordering doesn't matter for the feature itself. But indeed, it does for reproducibility. We could ensure we always reserialize it ordered, but it doesn't seem worth the complexity.

There are two fields that still use `HashSet`. A simple `sed` isn't quite enough, so I left those for now and we can tackle them separately.